### PR TITLE
Add commit log bootstrap tool to verify commit log fileset

### PR DESCRIPTION
**Pull request by robskillington**
cc @xichen2020 @prateek 

This will also help us investigate commit log issues by instrumenting how this tool executes on filesets we have encountered issues with.

### CODE REVIEW

File: tools/bootstrap_commitlogs/main/main.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+		debugListenAddress = *debugListenAddressArg
+	)
+
+	log := xlog.NewLogger(os.Stderr)
+
+	if debugListenAddress != "" {
+		go func() {
+			err := http.ListenAndServe(debugListenAddress, http.DefaultServeMux)
+			if err != nil {
+				log.Errorf("could not start debug listen server at '%s': %v", debugListenAddress, err)
```
Comment (prateek) [inline tools/bootstrap_commitlogs/main/main.go]: Maybe `Fatalf` here?

File: tools/bootstrap_commitlogs/main/main.go [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+	"github.com/m3db/m3db/x/io"
+	"github.com/m3db/m3x/instrument"
+	xlog "github.com/m3db/m3x/log"
+	"github.com/m3db/m3x/time"
+)
+
+var (
+	pathPrefixArg         = flag.String("path-prefix", "/var/lib/m3db", "Path prefix")
+	namespaceArg          = flag.String("namespace", "metrics", "Namespace")
+	blockSizeArg          = flag.String("block-size", "2h", "Block size")
```
Comment (prateek) [inline tools/bootstrap_commitlogs/main/main.go]: nit: `flag.Duration` instead of `flag.String`

File: tools/bootstrap_commitlogs/main/main.go [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+	flushSizeArg          = flag.Int("flush-size", 524288, "Flush size of commit log")
+	bootstrapRetentionArg = flag.String("retention", "48h", "Retention")
```
Comment (prateek) [inline tools/bootstrap_commitlogs/main/main.go]: nit: `flag.Duration` instead of `flag.String`

File: tools/bootstrap_commitlogs/main/main.go [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+			shardsAll = append(shardsAll, uint32(value))
+		}
+	} else if shardsCount > 0 {
+		for i := uint32(0); i < uint32(shardsCount); i++ {
+			rng := xtime.Range{Start: start, End: end}
+			shardTimeRanges[i] = xtime.NewRanges().AddRange(rng)
+			shardsAll = append(shardsAll, i)
+		}
+	} else {
+		flag.Usage()
```
Comment (prateek) [inline tools/bootstrap_commitlogs/main/main.go]: nit: mention explicitly the issue with bad shard configuration

File: tools/bootstrap_commitlogs/main/main.go [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+	}
+
+	retentionLen, err := time.ParseDuration(bootstrapRetention)
+	if err != nil {
+		log.Fatalf("could not parse retention '%s': %v", bootstrapRetention, err)
+	}
+
+	now := time.Now()
+	end := now.Truncate(blockSizeLen).Add(blockSizeLen)    // exclusive
+	start := now.Truncate(blockSizeLen).Add(-retentionLen) // inclusive
```
Comment (xichen2020) [inline tools/bootstrap_commitlogs/main/main.go]: Might as well name them `startInclusive` and `endExclusive` then?

File: tools/bootstrap_commitlogs/main/main.go [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+	var shardsAll []uint32
+	if strings.TrimSpace(shards) != "" {
+		for _, shard := range strings.Split(shards, ",") {
```
Comment (xichen2020) [inline tools/bootstrap_commitlogs/main/main.go]: Make `","` a const?

File: tools/bootstrap_commitlogs/main/main.go [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+			shard = strings.TrimSpace(shard)
+			if shard == "" {
+				continue
```
Comment (xichen2020) [inline tools/bootstrap_commitlogs/main/main.go]: Should this be an error instead?

**Comment by richardartoul** [comment]
@robskillington Once I'm done with the other two PR's, can I commandeer this diff? The code required to build this executable has changed, and I have a working version which I've been using to test my changes.

File: tools/bootstrap_commitlogs/main/main.go (hunk 4 above)
Comment (richardartoul) [reply tools/bootstrap_commitlogs/main/main.go]: What issue?

**Comment by richardartoul** [comment]
This PR has the updated code: https://github.com/m3db/m3db/pull/363

**Comment by robskillington** [comment]
Closing after @richardartoul implemented this with #363.

**Pull request closed without merging**